src/components/NewDatePicker/index.js

```diff
@@ -45,19 +44,11 @@ class NewDatePicker extends React.Component {
     }
 
     componentDidMount() {
-        if (this.props.autoFocus) {
-            this.showPicker();
-            this.textInputRef.focus();
+        if (!this.props.autoFocus) {
+            return;
         }
-
-        const shortcutConfig = CONST.KEYBOARD_SHORTCUTS.ESCAPE;
-        this.unsubscribeEscapeKey = KeyboardShortcut.subscribe(shortcutConfig.shortcutKey, () => {
-            if (!this.state.isPickerVisible) {
-                return;
-            }
-            this.hidePicker();
-            this.textInputRef.blur();
-        }, shortcutConfig.descriptionKey, shortcutConfig.modifiers, true, () => !this.state.isPickerVisible);
+        this.showPicker();
+        this.textInputRef.focus();
```

Comment:
Since we both show the picker and focus the input on component mount. Shouldn't we do the same on DOB? As suggested here https://github.com/Expensify/App/pull/16779#discussion_r1157491589 just for consistency. (whenever the calendar is shown, the input is also focused)